pkg/cluster/controllers/database/readiness.go

```diff
@@ -43,17 +43,28 @@ const (
 // - return postgre properties if postgre has available.
 func (p *PostgreSQLController) Readiness(ctx context.Context) (*lcm.CRStatus, error) {
 	var (
-		conn *Connect
-		err  error
+		conn   *Connect
+		client *pgx.Conn
+		err    error
 	)
 
 	name := p.HarborCluster.Name
 
-	conn, _, err = p.GetInClusterDatabaseInfo(ctx)
+	conn, client, err = p.GetInClusterDatabaseInfo(ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	defer client.Close(ctx)
+
+	if err := client.Ping(ctx); err != nil {
```

Comment:
We should not check the status with ping as the client in the operator process may not connect to psql instance deployed in another namespace.

Why do not use the status of psql CR?

@Ghostbaby

Reply:
done